OSSM-6615 Make sure IOR ignores services without selectors

luksa opened the pull request:
IOR finds a matching service for a pod by checking each service's label selector. However, this also matches services with an empty selector, which is wrong. These services should be ignored by IOR since an empty label selector indicates that the service endpoints are managed externally and not that the service should match all pods.

luksa committed:
OSSM-6615 Make sure IOR ignores services without selectors

IOR finds a matching service for a pod by checking each service's label selector. However, this also matches services with an empty selector, which is wrong. These services should be ignored by IOR since an empty label selector indicates that the service endpoints are managed externally and not that the service should match all pods.

jwendell reviewed (approved):
worth a unit test case for this?

luksa committed:
Add test

luksa commented:
/test integ-helm-2-5

luksa commented:
/retest

luksa commented:
/retest

luksa commented:
/retest

luksa commented:
/retest

luksa commented:
/retest

luksa committed:
Increase timeout

The previous timeout was only 3s, which caused the retry.UntilSuccessOrFail to fail on the first attempt instead of allowing the retry to be performed.

luksa commented:
/test maistra-istio-unit-2-5
